pymc/tests/test_distributions_timeseries.py

```diff
@@ -11,21 +11,85 @@
 #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
-
 import numpy as np
 import pytest
 
+from scipy import stats
+
+import pymc as pm
+
 from pymc.aesaraf import floatX
 from pymc.distributions.continuous import Flat, Normal
-from pymc.distributions.timeseries import AR, AR1, GARCH11, EulerMaruyama
+from pymc.distributions.timeseries import (
+    AR,
+    AR1,
+    GARCH11,
+    EulerMaruyama,
+    GaussianRandomWalk,
+    gaussianrandomwalk,
+)
 from pymc.model import Model
 from pymc.sampling import sample, sample_posterior_predictive
+from pymc.tests import test_distributions as td
 from pymc.tests.helpers import select_by_precision
+from pymc.tests.test_distributions_random import BaseTestDistributionRandom
+
+
+class TestGaussianRandomWalk(BaseTestDistributionRandom):
+    # Override default size for test class
+    size = None
+
+    pymc_dist = pm.GaussianRandomWalk
+    pymc_dist_params = {"mu": 1.0, "sigma": 2, "init": pm.Constant.dist(0), "steps": 4}
+    expected_rv_op_params = {"mu": 1.0, "sigma": 2, "init": pm.Constant.dist(0), "steps": 4}
+
+    checks_to_run = [
+        "check_pymc_params_match_rv_op",
+        "check_rv_inferred_size",
+    ]
+
+    def check_rv_inferred_size(self):
+        steps = self.pymc_dist_params["steps"]
+        sizes_to_check = [
+            None,
+            (),
+            1,
+            (1,),
+        ]
+        sizes_expected = [(steps + 1,), (steps + 1,), (1, steps + 1), (1, steps + 1)]
+        # sizes_to_check = [None, (), 1, (1,), 5, (4, 5), (2, 4, 2)]
+        # sizes_expected = [(), (), (1,), (1,), (5,), (4, 5), (2, 4, 2)]
+
+        for size, expected in zip(sizes_to_check, sizes_expected):
+            pymc_rv = self.pymc_dist.dist(**self.pymc_dist_params, size=size)
+            expected_symbolic = tuple(pymc_rv.shape.eval())
+            assert expected_symbolic == expected
+
+    def check_not_implemented(self):
+        with pytest.raises(NotImplementedError):
+            self.pymc_rv.eval()
+
+    # TODO: Move this out of this test class, not related, makes more sense together with 
+    # the other logp tests
+    def test_grw_inference(self):
```

Comment:
This is a pretty heavy test. Can we do with less steps, and a single chain?

Reply:
I reduced the steps, I think its only one chain as is?

Reply:
Oh do you mean one MCMC chain?

Reply:
Yeah one MCMC chain